### PR TITLE
Implement clean abort

### PR DESCRIPTION
**Pull request by aradi**
Environment can be be aborted in a clean way closing open files and
writing collected timing information.

Note: the error on failing scc-convergence has been converted into a warning (I think it is more appropriate), but code stops immediately stops after that.

### CODE REVIEW

File: prog/dftb+/lib_common/timerarray.F90 [hunk 1]
```diff
@@ -43,16 +46,43 @@ module timerarray
...
-  subroutine TTimerArray_init(this, timerItems)
+  subroutine TTimerArray_init(this, timerItems, maxLevel, header, unit)
 
     !> Instance
     type(TTimerArray), intent(out) :: this
 
     !> Names of the sub-timers to use
     type(TTimerItem), intent(in) :: timerItems(:)
 
+    !> Last level to be included (default: all levels are included)
```
Comment (bhourahine) [inline prog/dftb+/lib_common/timerarray.F90]: Last timer level to be included in printing / time evalualtion / ??? (default all ...

File: prog/dftb+/lib_common/environment.F90 [hunk 2]
```diff
@@ -36,7 +39,10 @@ module environment
     integer, public :: myGroup = 0
 
     !> Global timers
-    type(TTimerArray) :: globalTimer
+    type(TTimerArray), public :: globalTimer
+
+    !> Registry for files, which may be open and must be closed when environment is aborted
```
Comment (bhourahine) [inline prog/dftb+/lib_common/environment.F90]: Registry of files, ....

File: prog/dftb+/lib_common/environment.F90 [hunk 3]
```diff
@@ -48,13 +54,16 @@ module environment
...
   #:if WITH_MPI
-    procedure :: initMpi
+    procedure :: initMpi => TEnvironment_initMpi
   #:endif
   #:if WITH_SCALAPACK
-    procedure :: initBlacs
+    procedure :: initBlacs => TEnvironment_initBlacs
   #:endif
-
+
```
Comment (bhourahine) [inline prog/dftb+/lib_common/environment.F90]: wasn't spaces in previous

File: prog/dftb+/lib_common/environment.F90 [hunk 4]
```diff
@@ -96,21 +105,67 @@ module environment
...
+    class(TEnvironment), intent(inout) :: this
+
+    call this%globalTimer%writeTimings()
+    call this%fileFinalizer%closeAll()
+    flush(stdOut)
+
+  end subroutine TEnvironment_destruct
+    
+  
+  !> Gracefully aborts finalizing the current environment first
```
Comment (bhourahine) [inline prog/dftb+/lib_common/environment.F90]: Gracefully cleans up before aborting

File: prog/dftb+/lib_common/environment.F90 [hunk 5]
```diff
@@ -96,21 +105,67 @@ module environment
...
+  end subroutine TEnvironment_abort
+
+
+  !> Initlaizes the global timer of the environment.
+  subroutine TEnvironment_initGlobalTimer(this, timingLevel, header, unit)
+
+    !> Instance
+    class(TEnvironment), intent(inout) :: this
+
+    !> Timing level up to which timings should be printed (-1 for all)
```
Comment (bhourahine) [inline prog/dftb+/lib_common/environment.F90]: Manual may have to be revised if -1 prints all levels.

File: prog/dftb+/lib_common/environment.F90 [hunk 6]
```diff
@@ -96,21 +105,67 @@ module environment
 
 contains
 
-  !> Returns an initialized instance.
```
Comment (bhourahine) [inline prog/dftb+/lib_common/environment.F90]: Why the removal?

Comment (aradi) [reply prog/dftb+/lib_common/environment.F90]: Definitely not intentional.

File: prog/dftb+/lib_common/environment.F90 (hunk 4 above)
Comment (aradi) [reply prog/dftb+/lib_common/environment.F90]: Sure.

File: prog/dftb+/lib_common/environment.F90 (hunk 5 above)
Comment (aradi) [reply prog/dftb+/lib_common/environment.F90]: Done

File: prog/dftb+/lib_common/environment.F90 (hunk 3 above)
Comment (aradi) [reply prog/dftb+/lib_common/environment.F90]: Emacs... :wink: Removed.

File: prog/dftb+/lib_common/environment.F90 (hunk 2 above)
Comment (aradi) [reply prog/dftb+/lib_common/environment.F90]: Thx

**Comment by aradi** [comment]
Which other subroutine was loosing its documentation? I should put that back, but I couldn't find any others apart of the one which I've fixed now. (See commit 64d8ed9 ).

**Pull request merged**